File: 1537-MaximumScoreAfterSplittingAString/1537-MaximumScoreAfterSplittingAString.cpp
// Last updated: 7/24/2025, 8:26:46 AM
class Solution {
public:
    int maxScore(string s) {
        int numOnes =0;
        for(char &c: s)
            if(c=='1')
                numOnes++;
        int res=0,numZeros=0;

        for(int i=0;i<s.length()-1;i++){
            if(s[i]=='0')
                numZeros++;
            else 
                numOnes--;
            res = max(res,numOnes+numZeros);
        }
        return res;
    }
};

File: 1537-MaximumScoreAfterSplittingAString/1537-MaximumScoreAfterSplittingAString.py
# Last updated: 7/24/2025, 8:26:54 AM
class Solution:
    def maxScore(self, s: str) -> int:
        max_score = count_zeros_left = 0
        count_ones_right = s.count('1')

        for i in range(len(s) - 1):
            count_zeros_left += s[i] == '0'
            count_ones_right -= s[i] == '1'
            max_score = max(max_score, count_zeros_left + count_ones_right)

        return max_score